Ghidra/Features/WildcardAssembler/ghidra_scripts/FindInstructionWithWildcard.java

```diff
@@ -0,0 +1,346 @@
+	 * {@code WildAssemblyResolvedPatterns}. This created {@code AssemblyPatternBlock} does NOT have
+	 * any bits specified that are part of a wildcarded operand. This is in contrast to the original
+	 * {@code WildAssemblyResolvedPatterns} where those bits are specified to have the values that
+	 * correspond to the {@code WildOperandInfo} values found in the
+	 * {@code WildAssemblyResolvedPatterns}.
+	 */
+	class ReducedWildcardAssemblyResolvedPattern {
+		/**
+		 * The original WildAssemblyResolvedPatterns that this is based on
+		 */
+		WildAssemblyResolvedPatterns parent;
+		/**
+		 * The portion of the instruction that is NOT any wildcarded operand
+		 */
+		AssemblyPatternBlock maskedInstruction;
+
+		ReducedWildcardAssemblyResolvedPattern(WildAssemblyResolvedPatterns input) {
+			parent = input;
+
+			// Remove all the bits which correspond to wildcarded opcodes from the instruction and
+			// save the result as maskedInstruction
+			var reducedInstruction = input.getInstruction();
+			for (WildOperandInfo info : input.getOperandInfo()) {
+				reducedInstruction = reducedInstruction.maskOut(info.location());
+			}
+			maskedInstruction = reducedInstruction;
+		}
+
+		/**
+		 * Returns true of the given value shares the same {@code maskedInstruction} and wildcard(s)
+		 * as this instance.
+		 * 
+		 * @param other
+		 *            Value to compare against
+		 * @return True if both values share the same maskedInstruction and wildcard(s)
+		 */
+		boolean sameBaseEncoding(ReducedWildcardAssemblyResolvedPattern other) {
+			if (this.maskedInstruction.compareTo(other.maskedInstruction) != 0) {
+				return false;
+			}
+
+			for (WildOperandInfo info : this.parent.getOperandInfo()) {
+				var foundMatch = false;
+				for (WildOperandInfo otherInfo : other.parent.getOperandInfo()) {
+					if (info.wildcard().equals(otherInfo.wildcard())) {
+						if (!info.equals(otherInfo)) {
+							return false;
+						}
+						foundMatch = true;
+						break;
+					}
+				}
+				if (!foundMatch) {
+					return false;
+				}
+			}
+			return true;
+		}
+	}
+
+	/**
+	 * Searches for at most 10 matches for each given encoding, starting at {@code currentAddress}
+	 * and prints results to the console.
+	 * 
+	 * This searches encoding by encoding, restarting back at the start of memory for each.
+	 * 
+	 * Does not currently print wildcard information about the search results, but this could be
+	 * added.
+	 * 
+	 * @param encodings
+	 *            HashMap of encodings to that encoding's possible WildOperandInfo values.
+	 * @throws MemoryAccessException
+	 *             If we find bytes but can't read them
+	 */
+	private void searchMemoryForEncodings(
+			HashMap<AssemblyPatternBlock, HashSet<WildOperandInfo>> encodings,
+			ArrayList<WildAssemblyResolvedPatterns> allValidResolvedPatterns)
+			throws MemoryAccessException {
+
+		Memory memory = currentProgram.getMemory();
+
+		for (var entry : encodings.entrySet()) {
```

Comment:
I'm resetting the `matchCount` inside the outer loop (and the `searchFromAddress`) so I'll get up to 10 results for each of the encodings (rather than 10 total). There are certainly other ways to do it (and I'm happy to consider other suggestions) but my thought was to emphasize the different encodings in this demo as that's the "special sauce" of the WildcardAssembler and one of the concepts a dev needs to understand to use it. As a search tool, yes, having the search results in order regardless of encoding probably makes more sense, but my thought is the point of this script is more to show how to use the WildcardAssembler so other folks can make cool stuff. The great search tool is coming along with the other components in subsequent pull requests / Pickled Canary versions. Thoughts?